test/opt/eliminate_dead_functions_test.cpp

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+#include <vector>
+
+#include <gmock/gmock.h>
+
+#include "assembly_builder.h"
+#include "pass_fixture.h"
+#include "pass_utils.h"
+
+
+template <typename T> std::vector<T> concat(const std::vector<T> &a, const std::vector<T> &b) {
```

Comment:
This will change when I merge it with my other ongoing pull request: https://github.com/KhronosGroup/SPIRV-Tools/pull/815/.  I will not have this definition here; it will be moved to pass_utils.h.